src/python/pants/backend/python/goals/lockfile.py

```diff
@@ -156,20 +156,21 @@ def warn_python_repos(option: str) -> None:
 
 @dataclass(frozen=True)
 class _PipArgsAndConstraintsSetup:
+    resolve_config: ResolvePexConfig
     args: tuple[str, ...]
     digest: Digest
-    constraints: FrozenOrderedSet[PipRequirement]
-    only_binary: tuple[str, ...]
-    no_binary: tuple[str, ...]
 
 
 @rule_helper
-async def _setup_pip_args_and_constraints_file(resolve_name: str) -> _PipArgsAndConstraintsSetup:
-    args = []
-    digests = []
+async def _setup_pip_args_and_constraints_file(
+    resolve_name: str, *, use_pex: bool
```

Comment:
"pex" is pretty overloaded, so this parameter name might not be self-explanatory. And it pretty fundamentally changes the behavior of the function.

But... from context, it seems like this is "use pex to generate the lockfile". Given that that will soon (?) be the only way to generate a lockfile, the fundamental ignoring of the binary/constraints options is maybe ok. It might be good to point out that we explicitly don't support them for `poetry`, or eagerly fail when they're combined, maybe?